core/contenthash.go

```diff
@@ -103,7 +103,7 @@ func GetContentHashFromDef(
 		)
 		defer telemetry.End(span, func() error { return rerr })
 
-		dgst, err := bkcontenthash.Checksum(ctx, ref, "/", bkcontenthash.ChecksumOpts{}, nil)
+		dgst, err := bkcontenthash.Checksum(ctx, ref, subdir, bkcontenthash.ChecksumOpts{}, nil)
```

Comment:
:eyes: @sipsma looks like this was wrong, only discovered this when writing tests that accessed different sub-directories from within the same client+session.

Kinda surprised we haven't hit this anywhere else :thinking: are we hitting this code path as much as we should be?

Reply:
Yeah that looks like a typo, but probably didn't come up because until very recently I think all the uses of it ended up providing subdir as `/`. I could be missing something but from a quick glance I think the only situation now where subdir isn't `/` is when using module sources of `dir` kind, which is sort of obscure too.